client/network/src/behaviour.rs

```diff
@@ -181,23 +172,32 @@ impl<B: BlockT, H: ExHashT> Behaviour<B, H> {
 		role: Role,
 		user_agent: String,
 		local_public_key: PublicKey,
-		block_requests: block_requests::BlockRequests<B>,
-		finality_proof_requests: finality_requests::FinalityProofRequests<B>,
 		light_client_handler: light_client_handler::LightClientHandler<B>,
 		disco_config: DiscoveryConfig,
-		request_response_protocols: Vec<request_responses::ProtocolConfig>,
+		// Block and finality request protocol config.
+		block_request_protocol_config: request_responses::ProtocolConfig,
+		finality_request_protocol_config: request_responses::ProtocolConfig,
+		// All remaining request protocol configs.
+		mut request_response_protocols: Vec<request_responses::ProtocolConfig>,
```

Comment:
`block_request_protocol_config` and `finality_request_protocol_config` could be put inside `request_response_protocols` here. Are they kept seperate because they're necessary for the blockchain to function?

Reply:
When sending out block and finality requests, we need to know the names of those protocols. We can extract these names out of the corresponding `ProtocolConfig` structs.

The block request `ProtocolConfig` and the finality request `ProtocolConfig` are treated separately in the function arguments in order to

1. Extract the protocol names in the first lines of the function implementation.

2. To force callers of `Behaviour::new` to provide the two `ProtocolConfig`s as they are expected to be present.

With the above in mind, can you think of a way to make this less confusing @expenses?